spec/services/webpage_fetch_service_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe WebpageFetchService, type: :service do
+  describe '#call' do
+    context 'when querying a webpage returns success response' do
+      it 'returns an HTTParty Response' do
+        url = FFaker::Internet.http_url
+        WebMock.stub_request(:get, url)
+        result = described_class.new(url, nil).call
+
+        expect(result).to be_an_instance_of(HTTParty::Response)
+      end
+    end
+
+    context 'when querying a webpage with User-Agent' do
+      it 'returns an HTTParty Response' do
+        url = FFaker::Internet.http_url
+        user_agent = FFaker::FreedomIpsum.word
+        WebMock.stub_request(:get, url).with(headers: { 'User-Agent' => user_agent })
+        result = described_class.new(url, user_agent).call
+
+        expect(result).to be_an_instance_of(HTTParty::Response)
+      end
+    end
```

Comment:
When reading these two context, it's not clear what is the difference between using a User Agent and not using a user agent. 

This gives us 2 options:
- Either there is no much difference and then only 1 test is needed
- Either we are not asserting enough things — Is there the user agent in the headers of the response object? If yes we might want to test that // or maybe the returned HTML is difference 🤔 

I do not have the immediate solution but my point is just: Think of tests are a documentation, each context should clearly explain why it is different from the other contexts — i.e. why should I use `user_agent`, when is it ok to not provide the user agent, etc...

Reply:
I moved the `user-agent` out of this class as it does not belong here, added `headers` instead.